fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/mixin/attachment/AttachmentTargetsMixin.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.attachment;
+
+import java.util.IdentityHashMap;
+
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.entity.Entity;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.world.World;
+import net.minecraft.world.chunk.WorldChunk;
+
+import net.fabricmc.fabric.api.attachment.v1.AttachmentType;
+import net.fabricmc.fabric.impl.attachment.AttachmentSerializingImpl;
+import net.fabricmc.fabric.impl.attachment.AttachmentTargetImpl;
+
+@Mixin(value = {BlockEntity.class, Entity.class, World.class, WorldChunk.class})
+public class AttachmentTargetsMixin implements AttachmentTargetImpl {
+	@Unique
+	@Nullable
+	protected IdentityHashMap<AttachmentType<?>, Object> fabric_dataAttachments = null;
+
+	@SuppressWarnings("unchecked")
+	@Override
+	@Nullable
+	public <T> T getAttached(AttachmentType<T> type) {
+		return fabric_dataAttachments == null ? null : (T) fabric_dataAttachments.get(type);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	@Nullable
+	public <T> T setAttached(AttachmentType<T> att, @Nullable T value) {
```

Comment:
Nit: Id possibly rename `att` to `attachment`?

Reply:
probably even better to `type`